src/app/system/audit-trails/audit-trails.component.ts

```diff
@@ -0,0 +1,411 @@
+        map((entityName: string) => entityName ? this.filterEntityAutocompleteData(entityName) : this.auditTrailSearchTemplateData.entityNames)
+      );
+  }
+
+  /**
+   * Filters users.
+   * @param {string} userName User name to filter user by.
+   * @returns {any} Filtered users.
+   */
+  private filterUserAutocompleteData(userName: string): any {
+    return this.auditTrailSearchTemplateData.appUsers.filter((user: any) => user.username.toLowerCase().includes(userName.toLowerCase()));
+  }
+
+  /**
+   * Filters actions.
+   * @param {string} actionName Action name to filter action by.
+   * @returns {any} Filtered actions.
+   */
+  private filterActionAutocompleteData(actionName: string): any {
+    return this.auditTrailSearchTemplateData.actionNames.filter((action: any) => action.toLowerCase().includes(actionName.toLowerCase()));
+  }
+
+  /**
+   * Filters entities.
+   * @param {string} entityName Entity name to filter action by.
+   * @returns {any} Filtered entities.
+   */
+  private filterEntityAutocompleteData(entityName: string): any {
+    return this.auditTrailSearchTemplateData.entityNames.filter((entity: any) => entity.toLowerCase().includes(entityName.toLowerCase()));
+  }
+
+  /**
+   * Fetches Audit Trails data.
+   */
+  initAuditTrails() {
+    this.systemService.getAuditTrails(this.filterAuditTrailsBy, this.sort.active ? this.sort.active : '', this.sort.direction, 0, -1).subscribe((response: any) => {
+      this.auditTrailsData = response.pageItems;
+    });
+  }
+
+  /**
+   * Initialises and downloads the CSV file of the data.
+   */
+  downloadCSV() {
+    const replacer = (key: any, value: any) => value === undefined ? '' : value;
+    const header = ['Id', 'Resource Id', 'Status', 'Office', 'Made On', 'Maker', 'Checked On', 'Checker', 'Entity', 'Action', 'Client'];
+    const headerCode = ['id', 'resourceId', 'processingResult', 'officeName', 'madeOnDate', 'maker', 'checkedOnDate', 'checker', 'entityName', 'actionName', 'clientName'];
+    let csv = this.auditTrailsData.map((row: any) => headerCode.map(fieldName => JSON.stringify(row[fieldName], replacer)));
+    csv.unshift('data:text/csv;charset=utf-8,' + header.join(','));
+    csv = csv.join('\r\n');
+    const link = document.createElement('a');
+    link.setAttribute('href', encodeURI(csv));
+    link.setAttribute('download', 'Audit Trails.csv');
+    document.body.appendChild(link);
+    link.click();
+  }
+
+  /**
+   * Gets the date from the passed timestamp.
+   *
+   * TODO: Update once language and date settings are setup.
+   *
+   * @param {any} timestamp Timestamp from which date is to be extracted.
+   */
+  private getDate(timestamp: any) {
+    let day = timestamp.getDate();
+    let month = timestamp.getMonth() + 1;
+    const year = timestamp.getFullYear();
+    if (day < 10) {
+      day = `0${day}`;
+    }
+    if (month < 10) {
+      month = `0${month}`;
+    }
+    return `${year}-${month}-${day}`;
+  }
```

Comment:
Are you able to use the updated approach for date instead? https://github.com/openMF/web-app/wiki/Date-and-Language